### PR TITLE
Add licenses

### PR DESCRIPTION
**Pull request by Cypher1**

### CODE REVIEW

**Comment by Cypher1** [comment]
Should close #101

**Comment by bgogul** [comment]
@Cypher1 do you want us to review? (Some tests are failing, btw)

**Comment by Cypher1** [comment]
Could you have a look at the failing tests? I'll fix them today and then be ready for review.

**Comment by bgogul** [comment]
> Could you have a look at the failing tests? I'll fix them today and then be ready for review.

Are you not able to see the test results? In any case, I am surprised that the tests fail. They are passing at head. Let me retrigger it.

**Comment by Cypher1** [comment]
@bgogul I discovered it's due to some file path manipulations (that meant I hadn't updated filepaths when adding extensions, because without extensions it's hard to know which files need licenses or not). I've identified the files now and some of the files needed to be excluded (or we'd have to update the testdata and code generation).

See https://github.com/google-research/raksha/pull/653 for the new filters for selecting files that we expect to have licenses.

With that change, the new changes here should pass pre-commit and tests.

**Comment by Cypher1** [comment]
Fixed tests and opened https://github.com/google-research/raksha/pull/653 to prevent spurious pre-commit check warnings

File: src/analysis/souffle/epsilon_analysis.dl [hunk 1]
```diff
@@ -1,3 +1,18 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
```
Comment (markww) [inline src/analysis/souffle/epsilon_analysis.dl:2]: @asejfia added this file in 2022, right?

File: src/analysis/souffle/sensitivity_analysis.dl [hunk 2]
```diff
@@ -1,3 +1,18 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
```
Comment (markww) [inline src/analysis/souffle/sensitivity_analysis.dl:2]: This file added in 2022

File: src/analysis/souffle/sensitivity_analysis_unit_test_interface.dl [hunk 3]
```diff
@@ -1,3 +1,18 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
```
Comment (markww) [inline src/analysis/souffle/sensitivity_analysis_unit_test_interface.dl:2]: 2022

File: src/analysis/souffle/tests/arcs_fact_tests/diff_wrapper.sh [hunk 4]
```diff
@@ -1,4 +1,21 @@
 #!/bin/bash
+#
+# Copyright 2015 The Bazel Authors. All rights reserved.
```
Comment (markww) [inline src/analysis/souffle/tests/arcs_fact_tests/diff_wrapper.sh:3]: I don't think this is right, this is one of our files right?

File: src/common/logging/BUILD [hunk 5]
```diff
@@ -1,3 +1,20 @@
+#-------------------------------------------------------------------------------
+# Copyright 2022 Google LLC
```
Comment (markww) [inline src/common/logging/BUILD:2]: Just checking, is this 2022? Logging is old in the project, but I don't know about this `BUILD` file off the top of my head.

File: src/common/testing/BUILD [hunk 6]
```diff
@@ -1,3 +1,20 @@
+#-------------------------------------------------------------------------------
+# Copyright 2022 Google LLC
```
Comment (markww) [inline src/common/testing/BUILD:2]: Again, is this 2022? testing is one of the first things we added, but maybe this `BUILD` file came from a code reorg.

File: src/frontends/ambient/multimic.dl [hunk 7]
```diff
@@ -1,5 +1,18 @@
-// GENERATED FILE, DO NOT EDIT!
-
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
```
Comment (markww) [inline src/frontends/ambient/multimic.dl:3]: Is it right to replace the `GENERATED FILE` warning with the copyright? Also, should this file be here at all? That last question might be out of scope for this PR.

File: src/frontends/arcs/access_path.h [hunk 8]
```diff
@@ -1,3 +1,18 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
```
Comment (markww) [inline src/frontends/arcs/access_path.h:2]: I'm going to bet that this file is from 2021, but maybe it was moved. Similar for everything under `src/frontend/arcs`

File: src/analysis/souffle/tests/arcs_fact_tests/diff_wrapper.sh (hunk 4 above)
Comment (Cypher1) [reply src/analysis/souffle/tests/arcs_fact_tests/diff_wrapper.sh:3]: Sorry, not sure where I got the idea that this was from Bazel. Thanks

**Review by markww — APPROVED** [review]
LGTM once remaining comments are resolved.

**Comment by Cypher1** [comment]
Sorry @markww I realized the approach I used wasn't handling first check in date properly (nor understanding moved files).

I think I've fixed the affected files now, could you re-review?

File: src/frontends/arcs/xform_to_datalog/datalog_facts_test.cc [hunk 9]
```diff
@@ -64,10 +64,8 @@ INSTANTIATE_TEST_SUITE_P(
         std::make_tuple(ManifestDatalogFacts(),
                         *(AuthorizationLogicDatalogFacts::create(
                             utils::test::GetTestDataDir(path),
-                            "empty_auth_logic")),
-                        R"(// GENERATED FILE, DO NOT EDIT!
```
Comment (bgogul) [inline src/frontends/arcs/xform_to_datalog/datalog_facts_test.cc:68]: Not a big deal, but is there any specific reason to get rid of this comment?

**Review by bgogul — APPROVED** [review]

File: src/frontends/arcs/xform_to_datalog/datalog_facts_test.cc (hunk 9 above)
Comment (Cypher1) [reply src/frontends/arcs/xform_to_datalog/datalog_facts_test.cc:68]: Yes! I had mistaken some test files for source files. I think I've fixed them now, thanks for the reminder.

File: src/frontends/arcs/access_path.h (hunk 8 above)
Comment (Cypher1) [reply src/frontends/arcs/access_path.h:2]: Yep, they were moved. I can investigate each or just leave them with 2022?

Comment (Cypher1) [reply src/frontends/arcs/access_path.h:2]: @markww Sorry, I'm still unsure if you'd like for me to investigate these files to work out which can be licensed to 2021 or if 2022 is acceptable.

**Pull request closed without merging**